cmd/manager/main.go

```diff
@@ -210,3 +219,30 @@ func ensureDefaults(cfg *rest.Config, scheme *kruntime.Scheme) error {
 
 	return err
 }
+
+// generateMetrics creates a service, service monitor, starts a metrics server
+// and records the metrics to expose at the specified endpoint.
+
+func generateMetrics(cfg *rest.Config, namespace string) {
+
+	// Create a Service object to expose the metrics port.
+	service, err := metrics.ExposeMetricsPort(context.TODO(), customMetrics.MetricsPort)
+	if err != nil {
+		log.Errorf("Unable to expose metrics port: %v", err)
+	}
+
+	// Ensure that a nil pointer is not added to the slice.
```

Comment:
Generally I think adding a comment about the mechanics of the code is bad practice unless it is particularly unorthodox. I think in this case the purpose of checking for nil before generating the slice makes sense. If this is generally the only purpose, I think removing this comment makes sense here.


That being said if `service == nil` isn't it possible that we want to do something? Return an error perhaps? What state are we in if that happens? I think the same comment could be made just further above if `metrics.ExposeMetricsPort()` returns an error.

Reply:
If the service is nil, then we return an error stating `Unable to expose metrics port`. That part  is just to make sure that we do not pass a nil to the next function. Because if `nil` value is passed to `CreateServiceMonitor`, then a segfault occurs. So, when a service is not created, then servicemonitor is not created either and we do not expose metrics.

Reply:
I believe @kevinrizza is trying to point out that we may want to fail the operator rather than just logging the error and carrying on.